app/code/Magento/Inventory/Block/Adminhtml/Edit/Button/Delete.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace  Magento\Inventory\Block\Adminhtml\Edit\Button;
+
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\View\Element\UiComponent\Control\ButtonProviderInterface;
+
+/**
+ * Class Delete
+ */
+class Delete extends AbstractButton implements ButtonProviderInterface
+{
+    /**
+     * @return array
+     */
+    public function getButtonData()
+    {
+        $data = [];
+        if ($this->isEditMode()) {
+            $data = [
+                'label' => __('Delete Entry'),
+                'class' => 'delete',
+                'on_click' => 'deleteConfirm(\'' . __(
+                        'Are you sure you want to do this?'
+                    ) . '\', \'' . $this->getDeleteUrl() . '\')',
+                'sort_order' => 20,
+            ];
+        }
+        return $data;
+    }
+
+    /**
+     * @return string
+     */
+    private function getDeleteUrl()
```

Comment:
This method is not really needed, as it is used only once. You may inline this code in this case.